neard/tests/rpc_nodes.rs

```diff
@@ -1049,3 +1051,39 @@ fn test_check_tx_on_lightclient_must_return_does_not_track_shard() {
         });
     });
 }
+
+#[test]
+fn test_validators_by_epoch_id_current_epoch_not_fails() {
+    init_integration_logger();
+
+    let cluster = NodeCluster::new(1, |index| format!("validators_epoch_id{}", index))
+        .set_num_shards(1)
+        .set_num_validator_seats(1)
+        .set_num_lightclients(0)
+        .set_epoch_length(10)
+        .set_genesis_height(0);
+
+    cluster.exec_until_stop(|_genesis, _rpc_addrs, clients| async move {
+        let view_client = clients[0].1.clone();
+
+        spawn_interruptible(async move {
+            loop {
+                let final_block = view_client.send(GetBlock::latest()).await.unwrap().unwrap();
+
+                let res = view_client
+                    .send(GetValidatorInfo {
+                        epoch_reference: EpochReference::EpochId(EpochId(
+                            final_block.header.epoch_id,
+                        )),
+                    })
+                    .await;
+
+                if let Ok(Ok(validators)) = res {
+                    assert_eq!(validators.current_validators.len(), 1);
+                    System::current().stop();
+                }
+                sleep(std::time::Duration::from_millis(500)).await;
```

Comment:
Urgh, it feel really wrong to add 0.5 seconds to the CI time just to test for an edge case. I wonder if 

> "Test suite time is bounded by available CPU resources (in parciucular, tests do not rely on timeouts for sychronization)"

would be a good OKR for after we upgrade actix to the non-beta version.

Reply:
Oh, let's assume it's leftover code, I don't see a reason to have loop there.

Reply:
But we don't have other a way to ensure the blockchain has really started yet, I hope we'll be able to avoid it in future.

Reply:
Yeah, those things are always annoying and hard to design. Generally, you need some kind of a signal which can be emitted on readiness, and which you can listen to. That way, you can act immediately after the thing is ready, and not after some delay. This can either be some protocol level API (ie, there might be an rpc call is_ready, which returns the response only when the node is ready, whatever that means), or some internal observability. For example, we can pull the logs out out nodes until we noticed that all participants logged that they are ready.

Reply:
Don't think you need to sleep 500 milliseconds here. You can wait until block height is higher than 0 and then do the query.